features/FEATURE_BLE/ble/pal/PalSecurityManager.h

```diff
@@ -0,0 +1,1017 @@
+     */
+    virtual ble_error_t set_irk(
+        const irk_t &irk
+    ) = 0;
+
+    /**
+     * Set the local CSRK.
+     *
+     * @param[in] csrk signing key
+     * @retval BLE_ERROR_NONE On success, else an error code indicating reason for failure
+     */
+    virtual ble_error_t set_csrk(
+        const csrk_t &csrk
+    ) = 0;
+
+    /**
+     * Generate the Public key. This will also generate the private key.
+     * Public key will be returned as an event handler callback when it's ready.
+     *
+     * @retval BLE_ERROR_NONE On success, else an error code indicating reason for failure
+     */
+    virtual ble_error_t generate_public_key() = 0;
+
+    ////////////////////////////////////////////////////////////////////////////
+    // Authentication
+    //
+
+    /**
+     * Request authentication of the connection. This will trigger an appropriate reaction,
+     * which might include encryption or pairing/re-pairing.
+     *
+     * @param[in] connection connection handle
+     * @retval BLE_ERROR_NONE On success, else an error code indicating reason for failure
+     */
+    virtual ble_error_t request_authentication(
+        connection_handle_t connection
+    ) = 0;
+
+    /**
+     * Generate and return 8 octets of random data compliant with [FIPS PUB 140-2]
+     *
+     * @param[out] random_data returns 8 octets of random data
+     * @see BLUETOOTH SPECIFICATION Version 5.0 | Vol 2, Part H 2
+     * @retval BLE_ERROR_NONE On success, else an error code indicating reason for failure
+     */
+    virtual ble_error_t get_random_data(
+        random_data_t &random_data
+    ) = 0;
+
+    ////////////////////////////////////////////////////////////////////////////
+    // MITM
+    //
+
+    /**
+     * Set the default passkey that will be used when the SM needs a passkey to
+     * be displayed.
+     *
+     * By default, the pal security manager generates a random passkey when a
+     * passkey has to be displayed by the application. A call to this function
+     * with a valid passkey alter this behaviour and the SecurityManager shall
+     * pass the passkey set into SecurityManagerEvent::on_passkey_display .
+     *
+     * A call to this function with a null pointer will reset the behaviour and
+     * indicates to the security manager that passkeys passed to
+     * SecurityManagerEvent::on_passkey_display shall be randomly generated.
+     *
+     * @param[in] passkey Set the passkey that shall be used by the security
+     * manager when SecurityManagerEvent::on_passkey_display is called. If
+     * passkey is a null pointer then the security manager generates a random
+     * passkey every time it calls SecurityManagerEvent::on_passkey_display.
+     *
+     * @retval BLE_ERROR_NONE On success, else an error code indicating reason for failure
+     */
+    virtual ble_error_t set_display_passkey(
+        passkey_num_t passkey
```

Comment:
`passkey_num_t` is an uint32 so how do you pass the null pointer :)?

Reply:
updated comments, thanks